tsl/src/bgw_policy/retention_api.c

```diff
@@ -193,16 +193,18 @@ policy_retention_add(PG_FUNCTION_ARGS)
 			/* If all arguments are the same, do nothing */
 			ts_cache_release(hcache);
 			ereport(NOTICE,
-					(errmsg("retention policy already exists on hypertable \"%s\", skipping",
+					(errmsg("retention policy already exists for hypertable \"%s\", skipping",
 							get_rel_name(ht_oid))));
 			PG_RETURN_INT32(-1);
 		}
 		else
 		{
 			ts_cache_release(hcache);
-			elog(WARNING,
-				 "could not add retention policy due to existing policy on hypertable with "
-				 "different arguments");
+			ereport(WARNING,
+					(errmsg("retention policy already exists for hypertable \"%s\"",
+							get_rel_name(ht_oid)),
+					 errdetail("A policy already exists with different arguments."),
+					 errhint("Remove the existing policy before adding a new one.")));
 			PG_RETURN_INT32(-1);
```

Comment:
IMHO, I think this should be an error similar to how you get an error if you try to create a function that already exists. The execution "aborts" after this anyway (which is what I assume that the "return -1" is about), so an error is appropriate.